packages/web-components/src/components/ic-toast/ic-toast.tsx

```diff
@@ -0,0 +1,318 @@
+import successIcon from "../../assets/success-icon.svg";
+import warningIcon from "../../assets/warning-icon.svg";
+import { getSlot, isSlotUsed } from "../../utils/helpers";
+import {
+  IcActivationTypes,
+  IcStatusVariants,
+  IcThemeForegroundEnum,
+} from "../../utils/types";
+import { ActionAreaElementTypes } from "./ic-toast.types";
+
+const ICONS = {
+  neutral: {
+    icon: neutralIcon,
+    alt: "Neutral",
+  },
+  info: {
+    icon: infoIcon,
+    alt: "For your information",
+  },
+  warning: {
+    icon: warningIcon,
+    alt: "Warning",
+  },
+  error: {
+    icon: errorIcon,
+    alt: "Error",
+  },
+  success: {
+    icon: successIcon,
+    alt: "Success",
+  },
+};
+const AUTO_DISMISS_TIMER_REFRESH_RATE_MS = 10;
+const TOAST_HEADING_CHAR_LIMIT = 70;
+const TOAST_MESSAGE_CHAR_LIMIT = 140;
+
+/**
+ * @slot action - IcButton or IcLink is placed below header and message
+ * @slot neutral-icon - A custom neutral icon is placed on the left side of the component
+ */
+@Component({
+  tag: "ic-toast",
+  styleUrl: "ic-toast.css",
+  shadow: true,
+})
+export class Toast {
+  @Element() el: HTMLIcToastElement;
+  /**
+   * The title to display at the start of the toast. (NOTE: Should be no more than `70` characters)
+   */
+  @Prop() heading!: string;
+  /**
+   * The variant of the toast being rendered
+   */
+  @Prop() variant?: IcStatusVariants;
+  /**
+   * The main body message of the toast. (NOTE: Should be no more than `140` characters)
+   */
+  @Prop() message?: string;
+  /**
+   * How the toast will be dismissed. If manual will display a dismiss button.
+   */
+  @Prop() dismissMode?: IcActivationTypes = "manual";
+  /**
+   * If toast dismissMode is set to `automatic`, use this prop to define the time before the toast dismisses (in MILLISECONDS)
+   * (NOTE: Has a minimum value of `5000ms`)
+   */
+  @Prop() autoDismissTimeout? = 5000;
+  /**
+   * Provides a custom alt-text to be announced to screen readers, if slotting a custom neutral icon
+   */
+  @Prop() neutralIconAltText?: string;
+  /**
+   * Is emitted when the user dismisses the toast
+   */
+  @Event() toastDismiss: EventEmitter<void>;
+
+  @State() visible = false;
+  @State() timerProgress = 100;
+
+  dismissTimeout: number;
+  timerRefreshInterval: number;
+  isManual: boolean;
+  interactiveElements: ActionAreaElementTypes[] = [];
+
+  /**
+   * Used to display the individual toast
+   * @returns The element that previously had focus before the toast appeared
+   */
+  @Method()
+  async setVisible() {
+    if (!this.visible) this.visible = true;
+    if (!this.isManual) {
+      this.dismissTimeout = window.setTimeout(
+        this.dismissAction,
+        this.autoDismissTimeout
+      );
+      this.timerRefreshInterval = window.setInterval(
+        this.handleProgressChange,
+        AUTO_DISMISS_TIMER_REFRESH_RATE_MS
+      );
+      return null;
+    } else {
+      return document.activeElement as HTMLElement;
+    }
```

Comment:
Null is only returned when the toast is dismissed automatically. Since it can't be focused, there is no need to return whatever element previously had focus. There is probably a nicer way to write this without the branches though.